app/code/Magento/WishlistGraphQl/Model/Resolver/CustomerWishlistsResolver.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\WishlistGraphQl\Model\Resolver;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Wishlist\Model\ResourceModel\Wishlist\CollectionFactory;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+
+/**
+ * Fetches the Wishlists data according to the GraphQL schema
+ */
+class CustomerWishlistsResolver implements ResolverInterface
+{
+    /**
+     * @var CollectionFactory
+     */
+    private $_wishlistCollectionFactory;
+
+    /**
+     * @param CollectionFactory $wishlistCollectionFactory
+     */
+    public function __construct(CollectionFactory $wishlistCollectionFactory)
+    {
+        $this->_wishlistCollectionFactory = $wishlistCollectionFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $customerId = $context->getUserId();
+
+        /* Guest checking */
+        if (!$customerId && 0 === $customerId) {
```

Comment:
`0 === $customerId` is already covered by `!$customerId` check. Please make check more specific.